src/to_wiki.js

```diff
@@ -118,23 +168,48 @@ const serializer = new WikiTextSerializer({
    },
 
    text(state, node) {
-      let marks = node.marks || []
+      let marks = node.marks.map(m => m.type.name)
+
+      let firstNeedle = function(haystack, needles) {
+         return needles.reduce((min, needle) =>
+          Math.min(min, haystack.indexOf(needle)), haystack.length)
```

Comment:
`indexOf` will return `-1` if it can't find `needle` in `haystack`, which will be the `min` for the rest of this `reduce`. Do we guarantee that every `needle` is in `haystack`?